search/tests/test_indexing.py

```diff
@@ -245,3 +262,39 @@ def test_term_cache_without_data(self):
         a LearningResource isn't tagged with any terms.
         """
         self.thrice()
+
+    def test_vocab_cache_by_course(self):
+        """
+        Ensure that get_vocabs for one course doesn't
+        pollute the cache for another course.
+        """
+        # Create a resource in a separate course.
+        # pylint: disable=protected-access
+        self.course, course2 = copy_instance(self.course)
+        self.resource, resource2 = copy_instance(self.resource)
+        course2.run = self.course.run + " part 2"
```

Comment:
I like more `"{} part 2".format(self.course.run)`, but feel free to leave it as it is